packages/core/src/external/carequality/command/documents/send-ihe-gateway-results.ts

```diff
@@ -0,0 +1,128 @@
+import { Sequelize, QueryTypes } from "sequelize";
+import axios from "axios";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { RaceControl, checkIfRaceIsComplete, controlDuration } from "../../../../util/race-control";
+import { MetriportError } from "../../../../util/error/metriport-error";
+import { initSequelizeForLambda } from "../../../../util/sequelize";
+import { errorToString } from "../../../../util/error/shared";
+
+const REQUEST_ID_COLUMN = "request_id";
+
+const api = axios.create();
+
+dayjs.extend(duration);
+
+export const CONTROL_TIMEOUT = dayjs.duration({ minutes: 3 });
+const CHECK_DB_INTERVAL = dayjs.duration({ seconds: 5 });
+
+export const sendIHEGatewayResults = async <TableResult>({
+  requestId,
+  patientId,
+  cxId,
+  numOfGateways,
+  dbCreds,
+  endpointUrl,
+  resultsTable,
+}: {
+  requestId: string;
+  patientId: string;
+  cxId: string;
+  numOfGateways: number;
+  dbCreds: string;
+  endpointUrl: string;
+  resultsTable: string;
+}): Promise<void> => {
+  const sequelize = initSequelizeForLambda(dbCreds);
+
+  const raceControl: RaceControl = { isRaceInProgress: true };
+
+  try {
+    // Run the table count until it either times out, or all the results are in the database
+    const raceResult = await Promise.race([
+      controlDuration(
+        CONTROL_TIMEOUT.asMilliseconds(),
+        `IHE gateway reached timeout after ${CONTROL_TIMEOUT.asMilliseconds()} ms`
+      ),
+      checkIfRaceIsComplete(
+        () => isIheGatewayResultsComplete(sequelize, resultsTable, requestId, numOfGateways),
+        raceControl,
+        `IHE gateway results came back in full (${numOfGateways} links).`,
+        CHECK_DB_INTERVAL.asMilliseconds()
+      ),
+    ]);
+
+    const iheGatewayResults = await queryIHEGatewayResults<TableResult>(
+      sequelize,
+      resultsTable,
+      requestId
+    );
+
+    if (raceResult) {
+      console.log(
+        `${raceResult}. Got ${iheGatewayResults.length} successes out of ${numOfGateways} gateways for ${resultsTable}. RequestID: ${requestId}`
+      );
+      raceControl.isRaceInProgress = false;
+    }
+
+    await api.post(endpointUrl, {
+      requestId,
+      patientId,
+      cxId,
+      iheGatewayResults,
```

Comment:
What is the shape of this request? We should definitely not send the result of the DB directly to the API.

Reply:
Leaving this as unresolved until we align on the comment/thread about the generic `TableResult` type on `queryIHEGatewayResults`